OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -237,13 +246,19 @@ void ITick.Tick(Actor self)
 
 		protected virtual void Tick(Actor self)
 		{
+			if (productionTraits.Count > 0 && enabledProductionTraits.Count == 0)
```

Comment:
What is the purpose of `productionTraits.Count > 0` here?  `ProductionQueue` requires `Production` traits to live on the same actor, so won't work at all if there are none.  Immediately clearing the queue in this case seems reasonable to me.